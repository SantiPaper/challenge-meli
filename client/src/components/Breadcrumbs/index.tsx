import { Categories } from "../../types/items";
import { StyledBreadcrumbs } from "./style";

type Props = {
  categories: Categories;
};
export const Breadcrumbs = ({ categories }: Props) => {
  return (
    <StyledBreadcrumbs>
      <ul className="breadcrumbs">
        {categories.map((catergory) => (
          <li key={catergory} className="breadcrumbs__item">
            <a href="#" className="breadcrumbs__item__link">
              {catergory}
            </a>
          </li>
        ))}
      </ul>
    </StyledBreadcrumbs>
  );
};
