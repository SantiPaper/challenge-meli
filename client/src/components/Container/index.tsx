import { FC, PropsWithChildren } from "react";
import { StyledContainer } from "./style";

export const Container: FC<PropsWithChildren> = ({ children }) => {
  return <StyledContainer>{children}</StyledContainer>;
};
