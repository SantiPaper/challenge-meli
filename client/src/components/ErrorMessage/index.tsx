import { StyledErrorMessage } from "./style";

type Props = {
  error: string | null;
};

export const ErrorMessage = ({ error }: Props) => {
  return (
    <StyledErrorMessage>
      <h2 className="text">{error}</h2>
    </StyledErrorMessage>
  );
};
