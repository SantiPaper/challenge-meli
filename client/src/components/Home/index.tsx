import { StyledHome } from "./style";

export const Home = () => {
  return (
    <StyledHome>
      <h1 className="hero">
        Aun no realizaste tu b√∫squeda, te alentamos a que lo hagas!
      </h1>
    </StyledHome>
  );
};
