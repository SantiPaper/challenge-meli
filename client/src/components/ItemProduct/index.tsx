import { StyledItemProduct } from "./style";
import imgShipping from "../../../public/images/shipping.png";

type Props = {
  title: string;
  price: number;
  img: string;
  shipping: boolean;
  id: string;
};
export const ItemProduct = ({ title, price, img, shipping, id }: Props) => {
  return (
    <StyledItemProduct id={id}>
      <img
        className="product__img"
        src={img}
        alt={title}
        width={180}
        height={180}
      />

      <div className="product__info">
        <div className="product__info__price-container">
          <p className="product__info__price-container__price">$ {price}</p>
          {shipping && <img src={imgShipping} alt="" width={15} height={15} />}
        </div>
        <h3 className="product__info__title">{title}</h3>
      </div>
    </StyledItemProduct>
  );
};
