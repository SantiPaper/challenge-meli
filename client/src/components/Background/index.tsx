import { FC, PropsWithChildren } from "react";
import { Container } from "../Container";
import { StyledBackground } from "./style";

export const Background: FC<PropsWithChildren> = ({ children }) => {
  return (
    <Container>
      <StyledBackground>{children}</StyledBackground>
    </Container>
  );
};
