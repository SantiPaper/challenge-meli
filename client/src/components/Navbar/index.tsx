import { Container } from "../Container";
import { SearchInput } from "../SearchInput";
import { StyledNavbar } from "./style";
import { Link } from "react-router-dom";

export const Navbar = () => {
  return (
    <StyledNavbar>
      <Container>
        <div className="navbar__content">
          <Link to="/">
            <img
              className="navbar__img"
              width={70}
              src="/images/logo_meli.png"
              alt="Logo"
            />
          </Link>
          <SearchInput />
        </div>
      </Container>
    </StyledNavbar>
  );
};
