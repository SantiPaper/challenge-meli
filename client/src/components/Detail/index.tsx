import { useLocation } from "react-router-dom";
import { useFetch } from "../../hooks/useFetch";
import { useProductsContext } from "../../hooks/useProductsContext";
import { Breadcrumbs } from "../Breadcrumbs";
import { Loading } from "../Loading";
import { Background } from "../Background";
import { StyledDetail } from "./style";

export const Detail = () => {
  const location = useLocation();

  const { state, handleFormat } = useProductsContext();

  const product = state.product.item;

  const categories = state.product.categories;

  const { loading } = useFetch({
    status: "/",
    payload: location.pathname.split("/")[2],
  });

  return (
    <StyledDetail>
      {loading ? (
        <Loading />
      ) : (
        <>
          <Breadcrumbs categories={categories} />
          <Background>
            <div className="detail__info">
              <img
                className="detail__info__img"
                src={product.picture}
                alt={product.title}
                width={680}
                height={680}
              />

              <div className="detail__info__aside">
                <p className="detail__info__aside__condition">
                  {product.condition === "new" ? "Nuevo" : "Usado"}
                </p>
                <h1 className="detail__info__aside__title">{product.title}</h1>
                <p className="detail__info__aside__price">
                  {handleFormat(product.price.amount)}
                </p>
                <button className="detail__info__aside__button">Comprar</button>
              </div>
            </div>
            <div className="detail__description">
              <h3 className="detail__description__title">
                Descripci√≥n del producto
              </h3>
              <p className="detail__description__text">{product.description}</p>
            </div>
          </Background>
        </>
      )}
    </StyledDetail>
  );
};
