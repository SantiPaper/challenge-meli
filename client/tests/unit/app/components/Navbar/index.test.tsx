import { screen, render } from "@testing-library/react";
import { Navbar } from "../../../../../src/components/Navbar";
import { BrowserRouter } from "react-router-dom";

beforeEach(() => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
});

describe("Navbar", () => {
  test("Renderiza navbar", () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  test("Se encuentra el logo", () => {
    const logo = screen.getByAltText("Logo");
    expect(logo).toBeInTheDocument();
  });
});
