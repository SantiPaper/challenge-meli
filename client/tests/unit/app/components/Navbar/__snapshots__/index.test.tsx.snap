// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > Renderiza navbar 1`] = `
<div>
  <header
    class="sc-dmyCSP bligWw"
  >
    <div
      class="sc-beySPh gVYoAi"
    >
      <div
        class="navbar__content"
      >
        <a
          href="/"
        >
          <img
            alt="Logo"
            class="navbar__img"
            src="/images/logo_meli.png"
            width="70"
          />
        </a>
        <form
          class="sc-guDLey juwNII"
          data-testid="form"
        >
          <label
            class="sr-only"
            for="search"
          >
            Buscar productos
          </label>
          <input
            class="search__input"
            id="search"
            name="search"
            placeholder="Nunca dejes de buscar"
            type="search"
          />
          <button
            aria-label="Buscar"
            class="search__button"
            type="submit"
          >
            <img
              alt="Buscar"
              src="/images/search.png"
              width="20"
            />
          </button>
        </form>
      </div>
    </div>
  </header>
</div>
`;
